index.html

```diff
@@ -853,66 +868,227 @@ <h2 id="BaseAudioContext">
+                  attribute of the <a>AudioContext</a> is not already
+                  <code>running</code>:
+                    <ol>
+                      <li>Set the <a href="#widl-audiocontext-state">state</a>
+                      attribute of the <a>AudioContext</a> to
+                      <code>running</code>.
+                      </li>
+                      <li>Queue a task to fire a simple event named
+                      <code>statechange</code> at the <a>AudioContext</a>
+                      </li>
+                    </ol>
+                  </li>
+                </ol>
+              </li>
+            </ol>
           </dd>
           <dt>
             Promise&lt;void&gt; close()
           </dt>
           <dd>
-            Closes the audio context, releasing any system audio resources used
-            by the <a><code>BaseAudioContext</code></a>. This will not
-            automatically release all BaseAudioContext-created objects, unless
-            other references have been released as well; however, it will
-            forcibly release any system audio resources that might prevent
-            additional AudioContexts from being created and used, suspend the
-            progression of the <a><code>BaseAudioContext</code></a>'s <a href=
-            "#widl-BaseAudioContext-currentTime">currentTime</a>, and stop
-            processing audio data. The promise resolves when all
-            AudioContext-creation-blocking resources have been released. If
-            this is called on <a>OfflineAudioContext</a>, then return a promise
-            rejected with a <code>DOMException</code> whose name is
-            <code>InvalidStateError</code>.
+            Closes the <a>BaseAudioContext</a>, <a>releasing the system
+            resources</a> it's using. This will not automatically release all
+            AudioContext-created objects, but will suspend the progression of
+            the <a><code>AudioContext</code></a>'s <a href=
+            "#widl-AudioContext-currentTime">currentTime</a>, and stop
+            processing audio data.
+            <p>
+              <span class="synchronous">When close is called, execute these
+              steps:</span>
+            </p>
+            <ol>
+              <li>Let <em>promise</em> be a new Promise.
+              </li>
+              <li>If this <a>BaseAudioContext</a> is an
+              <a>OfflineAudioContext</a>, reject <em>promise</em> with
+              <code>NotSupportedError</code>, return it, and abort these steps.
+              </li>
+              <li>If the <em>control thread state</em> flag on the
+              <a>AudioContext</a> is <code>closed</code> reject the promise
+              with <code>InvalidStateError</code>, abort these steps, returning
+              <em>promise</em>.
+              </li>
+              <li>If the <a href="#widl-AudioContext-state">state</a> attribute
```

Comment:
This is to prevent a race across the control thread and the rendering thread when doing, say:

```
context.close();
context.suspend();
```

When calling suspend, the state is not "closed", because it's updated when the audio system has been released, per spec. We still want to reject the `suspend` call by throwing, so we use an internal flag for this.